cmd/cli/serve/serve.go

```diff
@@ -274,34 +275,30 @@ func serve(cmd *cobra.Command) error {
 		NodeInfoStoreTTL:      nodeInfoStoreTTL,
 		NetworkConfig:         networkConfig,
 	}
-
 	if isRequesterNode {
 		// We only want auto TLS for the requester node, but this info doesn't fit well
 		// with the other data in the requesterConfig.
 		nodeConfig.RequesterAutoCert = config.ServerAutoCertDomain()
 		nodeConfig.RequesterAutoCertCache = config.GetAutoCertCachePath()
-
 		cert, key := config.GetRequesterCertificateSettings()
-		nodeConfig.RequesterTLSCertificateFile = cert
-		nodeConfig.RequesterTLSKeyFile = key
+		nodeConfig.RequesterTLSCertificateFile, nodeConfig.RequesterTLSKeyFile, err = GetTLSCertificate(&nodeConfig, cert, key)
+		if err != nil {
+			return err
+		}
```

Comment:
How about including the previous line in the GetTLSCertificate function, so it becomes

```
cert, key, err := GetTLSCertificate(nodeConfig.RequesterSelfSigned)
if err != nil { ... } 

nodeConfig.RequesterTLSCert = cert
nodeConfig.RequesterTLSKey = key
```

the parameter passed to GetTLSCertificate can then be `shouldSelfSign bool`